internal/service/xray/resource_policy.go

```diff
@@ -0,0 +1,244 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package xray
+
+import (
+	"context"
+	"errors"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/xray"
+	awstypes "github.com/aws/aws-sdk-go-v2/service/xray/types"
+	"github.com/hashicorp/terraform-plugin-framework-timetypes/timetypes"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/hashicorp/terraform-provider-aws/internal/create"
+	"github.com/hashicorp/terraform-provider-aws/internal/errs"
+	"github.com/hashicorp/terraform-provider-aws/internal/framework"
+	fwflex "github.com/hashicorp/terraform-provider-aws/internal/framework/flex"
+	tfslices "github.com/hashicorp/terraform-provider-aws/internal/slices"
+	"github.com/hashicorp/terraform-provider-aws/internal/tfresource"
+	"github.com/hashicorp/terraform-provider-aws/names"
+)
+
+// @FrameworkResource("aws_xray_resource_policy", name="Resource Policy")
+func newResourceResourcePolicy(_ context.Context) (resource.ResourceWithConfigure, error) {
+	r := &resourceResourcePolicy{}
+
+	return r, nil
+}
+
+const (
+	ResNameResourcePolicy = "Resource Policy"
+)
+
+type resourceResourcePolicy struct {
+	framework.ResourceWithConfigure
+	framework.WithNoOpUpdate[resourceResourcePolicyData]
+}
+
+func (r *resourceResourcePolicy) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "aws_xray_resource_policy"
+}
+
+func (r *resourceResourcePolicy) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"policy_document": schema.StringAttribute{
```

Comment:
We can use `jsontypes.NormalizedType` from `github.com/hashicorp/terraform-plugin-framework-jsontypes/jsontypes` here to get rid of spurious JSON document diffs.